packages/manager/apps/catalog-revamp/src/hooks/useCatalog.ts

```diff
@@ -0,0 +1,100 @@
+import { useState, useEffect } from 'react';
+import { groupBy, includes, isEmpty, map, uniq } from 'lodash-es';
+
+export type Product = {
+  id: number;
+  name: string;
+  description: string;
+  lang: string;
+  categories: string[];
+  url: string;
+  regionTags: string[];
+  productName: string;
+  order: string;
+  universe: string;
+  category: string;
+};
+
+export const useCatalog = (catalogData: Product[] = []) => {
```

Comment:
You can pass in parameter categories, universes, text

useCatalog = ({categories, univers, text}) => {

useEffect(() => {
 /* do your */
do your task
setProducts(newProducts);
}, [categories])

useEffect(() => {
 /* do your */
do your task
setProducts(newProducts);
}, [univers])

useEffect(() => {
 /* do your */
do your task
setProducts(newProducts);
}, [text])

Reply:
Doing it with useCatalog({selectedCategories, selectedUniverses, searchText}) in Filters.tsx 👍

Reply:
Done